Examples/ObjectiveC/ManageOfflineRegionsExample.m

```diff
@@ -3,12 +3,281 @@
+    [self installConstraints];
+    
+}
+
+-(void)setupOfflinePackHandler {
+    // Setup offline pack notification handlers.
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(offlinePackProgressDidChange:) name:MGLOfflinePackProgressChangedNotification object:nil];
+}
+
+-(void)installConstraints {
+
+    NSArray *constraints = @[
+    [self.mapView.topAnchor
+     constraintEqualToAnchor:self.view.topAnchor],
+    [self.mapView.leadingAnchor
+     constraintEqualToAnchor:self.view.leadingAnchor],
+    [self.mapView.trailingAnchor
+     constraintEqualToAnchor:self.view.trailingAnchor],
+    [self.mapView.heightAnchor
+     constraintEqualToAnchor:self.view.heightAnchor multiplier:0.75],
+    [self.tableView.topAnchor
+     constraintEqualToAnchor:self.mapView.bottomAnchor],
+    [self.tableView.leadingAnchor
+     constraintEqualToAnchor:self.view.leadingAnchor],
+    [self.tableView.trailingAnchor
+     constraintEqualToAnchor:self.view.trailingAnchor],
+    [self.tableView.heightAnchor
+     constraintEqualToAnchor:self.view.heightAnchor multiplier:0.4],
+    [self.downloadButton.topAnchor
+     constraintEqualToAnchor:self.view.topAnchor constant:100],
+    [self.downloadButton.leadingAnchor
+     constraintEqualToAnchor:self.view.leadingAnchor constant:5],
+    [self.downloadButton.widthAnchor
+     constraintEqualToAnchor:self.view.widthAnchor multiplier:0.45],
+    [self.downloadButton.heightAnchor
+     constraintEqualToAnchor:self.downloadButton.widthAnchor multiplier:0.2]
+    ];
+    [NSLayoutConstraint activateConstraints:constraints];
+}
+
+/*
+  For the purposes of this example, remove any offline packs
+  that exist before the example is re-loaded.
+*/
+-(void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:true];
+    // When leaving this view controller, suspend offline downloads.
+    [[MGLOfflineStorage sharedOfflineStorage]
+     resetDatabaseWithCompletionHandler:^(NSError * _Nullable error) {
+        if (error != nil) {
+            // Handle the error here if packs can't be removed.
+            NSLog(@"Error: %@", error.localizedFailureReason);
+        } else {
+            // Start downloading.
+            [[MGLOfflineStorage sharedOfflineStorage] reloadPacks];
+        }
+    }];
+};
+
+- (void)startOfflinePackDownload: (UIButton *)sender {
+    // Setup offline pack notification handlers.
+    [self setupOfflinePackHandler];
+
+    /**
+     Create a region that includes the current map camera, to be captured
+     in an offline map. Note: Because tile count grows exponentially as zoom level
+     increases, you should be conservative with your `toZoomLevel` setting.
+     */
+    id <MGLOfflineRegion> region = [[MGLTilePyramidOfflineRegion alloc] initWithStyleURL:self.mapView.styleURL bounds:self.mapView.visibleCoordinateBounds fromZoomLevel:self.mapView.zoomLevel toZoomLevel: self.mapView.zoomLevel + 2];
```

Comment:
The convention to define variables in ObjC goes like:

```
<object type> <optional pointer><variable name> = <assignment>;
```

(The pointer `*` is only sometimes required depending on the type - Xcode will give you a helpful warning if its missing or needed though.)

In this case, you're specifying that `region` is going to be an `id` of conforming to the MGLOfflineRegion protocol. A cleaner way to do this would be to just use the type in the definition already:


```objc
MGLTilePyramidOfflineRegion *region = [[MGLTilePyramidOfflineRegion alloc] initWithStyleURL:self.mapView.styleURL
                                                                                     bounds:self.mapView.visibleCoordinateBounds
                                                                              fromZoomLevel:self.mapView.zoomLevel
                                                                                toZoomLevel: self.mapView.zoomLevel + 2];
```

In addition, in Obj-C the formatting convention is to align parameters by the colon `:` along new lines, as shown above.

Reply:
Thanks! When I try to align the region by colon, it looks WILD 😩